components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go

```diff
@@ -0,0 +1,260 @@
+		key := constants.POSTGRESQL + ip + strconv.Itoa(index)
+		nm.log.Debug("Call Initiated for Postgresql node having IP: ", ip)
+		go nm.MakeConcurrentCall(ip, constants.POSTGRESQL, reqBody.MountLocation, ch, key)
+		orderList = append(orderList, key)
+	}
+
+	for index, ip := range reqBody.OpensearchNodeIPs {
+		key := constants.OPENSEARCH + ip + strconv.Itoa(index)
+		nm.log.Debug("Call Initiated for Opensearch node having IP: ", ip)
+		go nm.MakeConcurrentCall(ip, constants.OPENSEARCH, reqBody.MountLocation, ch, key)
+		orderList = append(orderList, key)
+	}
+
+	for i := 0; i < len(orderList); i++ {
+		tempResponse := <-ch
+		for k, v := range tempResponse {
+			nfsMountResultMap[k] = v.MountResp
+			if v.MountResp.Error == nil {
+				shareMap[k] = v.MountLocRes
+				countMap[v.MountLocRes] = countMap[v.MountLocRes] + 1
+			}
+		}
+	}
+	nm.log.Debug(prettyMap(nfsMountResultMap))
+	nm.log.Debug("All calls Completed")
+
+	MakeRespBody(respBody, countMap, orderList, nfsMountResultMap, shareMap, reqBody.MountLocation)
+	return respBody
+}
+
+func prettyMap(mp map[string]models.NFSMountResponse) string {
+	b, _ := json.MarshalIndent(mp, "", "  ")
+	return string(b)
+}
+
+func (nm *NFSMountService) MakeConcurrentCall(ip string, nodeType string, mountLocation string, respChan chan map[string]TempResponse, key string) {
+	res, err := nm.DoAPICall(ip, mountLocation)
+	nm.log.Debugf("result got from /nfs-mount-loc API for %s: %v", ip, res)
+
+	mountResp := prepareMountResp(ip, nodeType, mountLocation, res, err)
+
+	respMap := make(map[string]TempResponse)
+	tempResp := TempResponse{
+		MountResp: mountResp,
+	}
+	if err == nil {
+		tempResp.MountLocRes = *res
+	}
+	respMap[key] = tempResp
+	respChan <- respMap
+}
+
+func (nm *NFSMountService) DoAPICall(ip string, mountLocation string) (*models.NFSMountLocResponse, error) {
+	reqURL := fmt.Sprintf("http://%s:%s%s", ip, nm.port, constants.NFS_MOUNT_LOC_API_PATH)
+	nm.log.Debug("Request URL: ", reqURL)
+
+	reqBody := models.NFSMountLocRequest{
+		MountLocation: mountLocation,
+	}
+
+	// Converting Body into json encoding for passing into request
+	reqBodyJSON, err := json.Marshal(reqBody)
+	if err != nil {
+		nm.log.Error(err.Error())
+		return nil, errors.New("Failed to Marshal: " + err.Error())
+	}
+
+	httpReq, err := http.NewRequest(http.MethodPost, reqURL, bytes.NewBuffer(reqBodyJSON))
+	if err != nil {
+		nm.log.Error(err.Error())
+		return nil, errors.New("Failed to Create HTTP request: " + err.Error())
+	}
+
+	client := http.Client{
+		Timeout: constants.TIMEOUT * time.Second,
+	}
+
+	// Making the actual call
+	resp, err := client.Do(httpReq)
+	if err != nil {
+		nm.log.Error(err.Error())
+		return nil, errors.New("Failed to send the HTTP request: " + err.Error())
+	}
+
+	return nm.GetResultStructFromRespBody(resp.Body)
+}
+
+func (nm *NFSMountService) GetResultStructFromRespBody(respBody io.Reader) (*models.NFSMountLocResponse, error) {
+	body, err := ioutil.ReadAll(respBody) // nosemgrep
```

Comment:
Can we find a solution for this rather using //nosemgrep? Use below :- respBody, err := 
respBody, err := io.ReadAll(resp.Body)

Reply:
As of now, we are using 1.15 golang version and golang change this in 1.16. so we need to use this flag only. Once we upgrade the golang then only we can use the above code.